server/pulp/server/db/model/__init__.py

```diff
@@ -983,6 +983,24 @@ class CeleryBeatLock(AutoRetryDocument):
     _ns = StringField(default='celery_beat_lock')
 
 
+class ResourceManagerLock(AutoRetryDocument):
+    """
+    Single document collection which gives information about the current resource manager lock.
+
+    :ivar name: string representing the resource manager instance name
+    :type name: basestring
+    :ivar lock: A unique key set to "locked" when lock is acquired.
+    :type lock: basestring
+    :ivar _ns: (Deprecated), Contains the name of the collection this model represents
+    :type _ns: mongoengine.StringField
+    """
+    name = StringField(required=True)
+    lock = StringField(required=True, default="locked", unique=True)
+
+    # For backward compatibility
+    _ns = StringField(default='resource_manager_lock')
```

Comment:
We need to continue modeling this because it gets injected onto every document. This is configured at the PyMongo layer [here](https://github.com/pulp/pulp/blob/644b607528c406fddc69b2c07fa491c2497235a4/server/pulp/server/db/connection.py#L159). We will remove the son_injector and add a migration to remove the field from each collection with [issue 625](https://pulp.plan.io/issues/625) as part of Pulp 3.